### PR TITLE
Issue #2459: updated clean_tables to run in batches of 1000 rows

### LINKED ISSUE #2459: clean_tables script to prepare a database for import can not handle too many rows in a particular table
### Current Behavior

When running the clean_tables script to prepare a database for import, the process gets killed if there are too many rows in a particular table.  

### Expected Behavior

The clean_tables script to prepare a database for import, the process is successful regardless of the number of rows in a table.

### Steps to Reproduce

1. Execute clean_tables script table with large table

### Environment

Staging

### Additional Context

Peer Dave Kennedy: 

- In order to fix, delete rows should be batched in groups of 1000.
- Incorporate this code directly into the clean_tables script:

def delete_objects_in_batches(model):
    total_deleted = 0
    while True:
        pks = list(model.objects.values_list('pk', flat=True)[:BATCH_SIZE])
        if not pks:
            break
        with transaction.atomic():
            deleted, _ = model.objects.filter(pk__in=pks).delete()
            total_deleted += deleted
        [logger.info](http://logger.info/)(f"Deleted {deleted} objects, total deleted: {total_deleted}")
        sleep(0.1)
    [logger.info](http://logger.info/)(f"Finished deleting. Total deleted: {total_deleted}")

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by dave-kennedy-ecs**
## Ticket

Resolves #2459 

## Changes

- clean_tables script modified to delete rows from tables in batches of 1000

## Context for reviewers

clean_tables is run as the second step of 3 in export_tables, clean_tables, import_tables.  For more info, refer to the docs\operations\import_export.md documentation on how to run

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/management/commands/clean_tables.py [hunk 1]
```diff
@@ -56,14 +56,24 @@ def handle(self, **options):
...
+
+        Delete in batches to be able to handle large tables"""
         try:
             # Get the model class dynamically
             model = apps.get_model("registrar", table_name)
-            # Use a transaction to ensure database integrity
-            with transaction.atomic():
-                model.objects.all().delete()
-            logger.info(f"Successfully cleaned table {table_name}")
+            BATCH_SIZE = 1000
```
Comment (therealslimhsiehdy) [inline src/registrar/management/commands/clean_tables.py:65]: Suggestion for code below -- my reasoning is that I'm not the biggest fan of `while True:` as we could have an infinite loop:

```
try:
        # Get the model class dynamically
        model = apps.get_model("registrar", table_name)
        BATCH_SIZE = 1000
        total_deleted = 0
        
        # Get initial batch of primary keys
        pks = list(model.objects.values_list("pk", flat=True)[:BATCH_SIZE])
        
        # Loop until no more primary keys are returned
        while pks:
            # Use a transaction to ensure database integrity
            with transaction.atomic():
                deleted, _ = model.objects.filter(pk__in=pks).delete()
                total_deleted += deleted
            logger.debug(f"Deleted {deleted} {table_name}s, total deleted: {total_deleted}")
            
            # Get next batch of primary keys
            pks = list(model.objects.values_list("pk", flat=True)[:BATCH_SIZE])
        
        logger.info(f"Successfully cleaned table {table_name}, deleted {total_deleted} rows")
    except LookupError:
        logger.error(f"Model for table {table_name} not found.")
    except Exception as e:
        logger.error(f"Error cleaning table {table_name}: {e}")
```

File: src/registrar/tests/test_management_scripts.py [hunk 2]
```diff
@@ -33,6 +33,10 @@
 logger = logging.getLogger(__name__)
 
 
+class CustomDeleteException(Exception):
+    pass
```
Comment (therealslimhsiehdy) [inline src/registrar/tests/test_management_scripts.py:37]: **Question:** Is this pass on purpose? If so, what is the reasoning?

**Review by therealslimhsiehdy — COMMENTED** [review]
Overall looks good, I left a code suggestion comment and question (will test locally and/or on Meoward tomorrow)

**Review by therealslimhsiehdy — APPROVED** [review]
Was able to export and then clean tables, but ran into an issue with importing and let Dave know. Regardless, the important part of this PR is cleaning the table in batches and that worked great. 

Screenshot 2024-07-30 at 2 11 10 PM

Also just want to confirm: When exporting and then cleaning table, this cleans the User table meaning I have also removed myself and therefore should not able to access admin, correct?

File: src/registrar/management/commands/clean_tables.py (hunk 1 above)
Comment (abroddrick) [reply src/registrar/management/commands/clean_tables.py:65]: ++ for avoiding while true, though wouldn't block on it fully

**Comment by dave-kennedy-ecs** [comment]
> Was able to export and then clean tables, but ran into an issue with importing and let Dave know. Regardless, the important part of this PR is cleaning the table in batches and that worked great.
> 
> Screenshot 2024-07-30 at 2 11 10 PM
> 
> Also just want to confirm: When exporting and then cleaning table, this cleans the User table meaning I have also removed myself and therefore should not able to access admin, correct?

That is correct.  You may need to run UserFixture.load() to reload the users.

File: src/registrar/tests/test_management_scripts.py (hunk 2 above)
Comment (dave-kennedy-ecs) [reply src/registrar/tests/test_management_scripts.py:37]: This is no longer needed.  I had added a custom exception and then was testing for it, but found this was not necessary to do.

I deleted it.

File: src/registrar/management/commands/clean_tables.py (hunk 1 above)
Comment (dave-kennedy-ecs) [reply src/registrar/management/commands/clean_tables.py:65]: Ok, I made this change

**Pull request merged**